lib/jekyll/theme.rb

```diff
@@ -43,7 +47,7 @@ def path_for(folder)
     end
 
     def realpath_for(folder)
-      File.realpath(Jekyll.sanitized_path(root, "_#{folder}"))
+      File.realpath(Jekyll.sanitized_path(root, folder.to_s))
```

Comment:
Special casing is a non-starter for me in this already complicated codebase. I want a way to fetch the path on disk for a directory. If the directory starts with a `_`, then let's ask for it that way. It's way more direct to ask for `_sass` and get `_sass` then to ask for `sass` and get `_sass`.

@spudowiar The difference here is that this uses `File.realpath` which resolves symlinks. `Site#in_theme_dir` does not.